### PR TITLE
[Completion] Skip `tryOptimizeGenericDisjunction` if there's a completion child

### PR DESCRIPTION
**Pull request by hamishknight**
Attempting to favor a disjunction choice here can lead to unhelpful results if there's e.g a code completion token argument, since it acts as a placeholder.

rdar://127844278

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please SourceKit stress test

File: lib/Sema/ConstraintSystem.cpp [hunk 1]
```diff
@@ -6658,6 +6658,16 @@ static bool shouldHaveDirectCalleeOverload(const CallExpr *callExpr) {
 }
 #endif
 
+ASTNode ConstraintSystem::includingParentApply(ASTNode node) {
+  if (auto *expr = getAsExpr(node)) {
+    if (auto apply = dyn_cast_or_null<ApplyExpr>(getParentExpr(expr))) {
```
Comment (xedin) [inline lib/Sema/ConstraintSystem.cpp:6663]: Nit: can replace `dyn_cast_or_null` with `getAsExpr<ApplyExpr>`

**Review by xedin — APPROVED** [review]

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
Stress tester run: https://ci.swift.org/job/swift-PR-macos-with-sourcekit-stress-tester/358/

**Comment by hamishknight** [comment]
@swift-ci please SourceKit stress test

**Comment by hamishknight** [comment]
Stress tester result is UPASS

**Pull request merged**